source/common/common/thread.cc

```diff
@@ -21,7 +26,13 @@ Thread::Thread(std::function<void()> thread_routine) : thread_routine_(thread_ro
   UNREFERENCED_PARAMETER(rc);
 }
 
-int32_t Thread::currentThreadId() { return syscall(SYS_gettid); }
+int32_t Thread::currentThreadId() {
+#ifdef __linux__
+  return syscall(SYS_gettid);
+#elif defined(__APPLE__)
+  return pthread_mach_thread_np(pthread_self());
```

Comment:
Any reason for not using `pthread_threadid_np()` here?

    uint64_t tid;
    pthread_threadid_np(NULL, &tid);